aml-development/frontend
# feat(listing mgmt): table export csv

AMLOS-802

**Description**
As a user I would like to be able to export the Listing Management table to a file that can be opened by excel.

**Acceptance Criteria**
- A user has a button in Listing Management tables that will allow them to export the table to an excel capable format
- The export should include all data from all pages, not just what's visible on one screen.
- Columns that are hidden by the user can be ignored.

**How To Test**
Pull this branch
Go to Listing Management and click on the Table View button

- Verify user has an Export button to the right of the search bar.  Clicking that button has two options (download the current page as CSV vs download all pages as CSV).
- Clicking the download current page option exports only the currently visible listings on the table.  You can verify this by clicking next/prev and seeing if that option still downloads only the current page.
- Clicking the download all pages option exports all pages.  Note that this preserves all filters that do not count towards pagination (ie section filter, such as my listings filtering down to only your listings, status filter, org filter, enabled/disabled filter, etc).  If the count on your current filter shows X listings, the export should have X listings as well.
- If you show/hide columns in the table, hidden columns are ignored in the CSV export.
